src/Core/Resolvers/DWSqlQueryBuilder.cs

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Data.Common;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Models;
+using Microsoft.Data.SqlClient;
+
+namespace Azure.DataApiBuilder.Core.Resolvers
+{
+    /// <summary>
+    /// Class for building DwSql queries.
+    /// </summary>
+    public class DWSqlQueryBuilder : BaseSqlQueryBuilder, IQueryBuilder
+    {
+
+        private static DbCommandBuilder _builder = new SqlCommandBuilder();
+
+        /// <inheritdoc />
+        public override string QuoteIdentifier(string ident)
+        {
+            return _builder.QuoteIdentifier(ident);
+        }
+
+        /// <inheritdoc />
+        public string Build(SqlQueryStructure structure)
+        {
+            string columns;
+            string stringAgg = string.Empty;
+            int i = 0;
+            // Iterate through all the columns and build the string_agg
+            foreach (LabelledColumn column in structure.Columns)
+            {
+                // Generate the col value.
+                bool subQueryColumn = structure.IsSubqueryColumn(column);
+                string col_value = column.Label;
+
+                // If the column is not a subquery column and is not a string, cast it to string
+                if (!structure.IsSubqueryColumn(column) && structure.GetColumnSystemType(column.ColumnName) != typeof(string))
+                {
+                    col_value = $"CAST({col_value} AS NVARCHAR(MAX))";
+                }
+
+                // Create json. Example: "book.title": "Title" would be a sample output.
+                stringAgg += $"\"{column.Label}\":\"\' + STRING_ESCAPE(ISNULL({col_value},''),'json') + \'\"";
+                i++;
+
+                // Add comma if not last column. example: {"id":"1234","name":"Big Company"}
+                // the below ensures there is a comman after id but not after name.
+                if (i != structure.Columns.Count)
+                {
+                    stringAgg += ",";
+                }
+            }
+
```

Comment:
NIT: can we add the sample content of stringAgg as code comment here for better understanding?

Reply:
Added.